crates/ruff_python_formatter/src/trivia.rs

```diff
@@ -279,7 +306,15 @@ impl<'a> SimpleTokenizer<'a> {
             '\\' => TokenKind::Continuation,
 
             c => {
-                let kind = TokenKind::from_non_trivia_char(c);
+                let kind = if self.eat_multichar("if", c) {
```

Comment:
I believe this will incorrectly lex `ify` as an `if` keyword because it doesn't test that a whitespace directly follows it. 

I'm also unsure if LLVM is able to short-circuit if the expression, for example, starts with a number.

I think the way I would implement this is to implement lexing for identifiers, and then use a match on the lexed identifier to test if it is a keyword. You'll need to add a dependency to `unic-ucd-ident` (used by RustPython's lexer) to detect the start/end of an identifier.

```rust
if is_identifier_start(c) {
	self.cursor.eat_while(is_identifier_continuation);

	// If we want to support all identifiers, then we need to handle the string prefixes `b'`, `rf` etc. but we can ignore those for keywords
	
	let range = TextRange::at(self.offset, token_len);
	let kind = match &self.source[range] {	 // This requires storing the whole source on the tokenizer. I think that's fine
		"if" => TokenKind::If,
		"else" => TokenKind::Else,
		"match" => TokenKind::Match // Match is a soft keyword that depends on the context but we can always lex it as a keyword and leave it to the caller (parser) to decide if it should be handled as an identifier or keyword.
		...,
		_ => TokenKind::Other // Potentially an identifier, but only if it isn't a string prefix. We can ignore this for now https://docs.python.org/3/reference/lexical_analysis.html#string-and-bytes-literals
	}
}

fn is_identifier_start(c: char) -> bool {
	c.is_ascii_alphabetic() || c == '_' || is_non_ascii_identifier_start(c) 
}

// Checks if the character c is a valid continuation character as described
// in https://docs.python.org/3/reference/lexical_analysis.html#identifiers
fn is_identifier_continuation(c: char) -> bool {
    if c.is_ascii() {
        matches!(c, 'a'..='z' | 'A'..='Z' | '_' | '0'..='9')
    } else {
        is_xid_continue(c)
    }
}

fn is_non_ascii_identifier_start(c: char) -> bool {
    is_xid_start(c)
}
```

Something similar should work for the backward parsing, except that it must start with an `is_non_ascii_identifier_start` and the last (or first, depending on how you look at the problem) must be `is_identifier_start` . We should be able to move the match expression into some shared function.